packages/@aws-cdk/toolkit/lib/actions/deploy/index.ts

```diff
@@ -260,4 +260,12 @@ export interface DeployOptions extends BaseDeployOptions {
    * @deprecated Implement in IoHost instead
    */
   readonly progress?: StackActivityProgress;
+
+  /**
+   * Represents configuration property overrides for hotswap deployments.
+   * Currently only supported by ECS.
+   *
+   * @default - no overrides
+   */
+  readonly hotswapProperties?: HotswapPropertyOverrides;
```

Comment:
`HotswapPropertyOverrides` and `EcsHotswapProperties` are currently classes.

It's done quite weirdly at the moment, but I think for the public interface we need to turn this into structs and than create the classes from the struct input, similar like it was done before.